sql/core/src/main/scala/org/apache/spark/sql/execution/command/functions.scala

```diff
@@ -82,8 +82,11 @@ case class CreateFunctionCommand(
     } else {
       // Handles `CREATE OR REPLACE FUNCTION AS ... USING ...`
       if (replace && catalog.functionExists(func.identifier)) {
-        // alter the function in the metastore
-        catalog.alterFunction(func)
+        // replace the function in the metastore
+        // as of now Hive only alter name, owner, class name, type but not resource URI
+        // So in order to replace function spark needs to delete and create the function
+        catalog.dropFunction(func.identifier, ignoreIfExists)
+        catalog.createFunction(func, ignoreIfExists)
```

Comment:
AFAIK, this can cause authorization issue when the users have only `CREATE` or `ALTER` privilege.

Reply:
fixed, if the location is unchanged then alterFunction is called